### PR TITLE
Added local node_module check for NodePackageAnalyzer

### PR DESCRIPTION
**Pull request by yarisvt**
## Description of Change

Ignore packages that start with "./", since those are local packages
## Have test cases been added to cover the new functionality?

*no*

### CODE REVIEW

**Comment by jeremylong** [comment]
Do you have an example where `version.matches("^[.~]*/.*")` indicates a local node module is being used?

**Comment by yarisvt** [comment]
@jeremylong 

I am using vaadin to build a web app and it automatically generates js files in the `node_modules` folder. It then puts the following local path in the package.json (without prefixing it with `file:`)
image


According to the [npm docs](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#local-paths), local paths can use four forms:
```
../foo/bar
~/foo/bar
./foo/bar
/foo/bar
```

Therefore, the regex `^[.~]*/.*` matches all four of them:
image

File: core/src/main/java/org/owasp/dependencycheck/analyzer/NodePackageAnalyzer.java [hunk 1]
```diff
@@ -297,10 +297,10 @@ public static boolean shouldSkipDependency(String name, String version, boolean
 
         // this seems to produce crash sometimes, I need to tests
         // using a local node_module is not supported by npm audit, it crash
-        if (Objects.nonNull(version) && version.startsWith("file:")) {
+        if (Objects.nonNull(version) && (version.startsWith("file:") || version.matches("^[.~]*/.*"))) {
```
Comment (jeremylong) [inline core/src/main/java/org/owasp/dependencycheck/analyzer/NodePackageAnalyzer.java]:
```suggestion
        if (Objects.nonNull(version) && (version.startsWith("file:") || version.matches("^[.~]{0,2}/.*"))) {
```

**Review by jeremylong — APPROVED** [review]

**Comment by jeremylong** [comment]
Thanks for the detailed response! I made one very minor change to the regex (s/\*/{0,2}/).

**Comment by yarisvt** [comment]
Great!

**Pull request merged**